### PR TITLE
Cleaning up the defining a component page

### LINKED ISSUE #602: General review of the Guides
The Guides are long overdue for a top-to-bottom review. Let's split up the work and go through each section, looking for:

* clarity of writing;
* plenty of examples to illustrate concepts;
* starting with simple, easy concepts and moving to advanced, difficult concepts;
* keeping each page as short as possible and splitting up long pages;
* focus on the "happy path", and avoidance of edge cases;
* liberal linking to the API docs;
* removing mention of controllers as much as possible;
* and generally following the [Style Guide](https://github.com/emberjs/guides/blob/master/CONTRIBUTING.md).

Here are the sections and who is taking which:

* [x] Getting Started (will be taken care of via the tutorial)
* [x] ~~The Object Model~~ (recently reviewed)
* [x] Templates (@michaelrkn)
* [x] Routing (@michaelrkn)
* [x] Components (@michaelrkn)
* [x] Controllers (@michaelrkn)
* [x] Models (@bmac)
* [x] Services (@michaelrkn)
* [x] Testing (@toddjordan)
* [x] Ember Inspector (@mattw59)
* [x] ~~Addons and Dependencies~~ (moved to #824)
* [x] ~~Configuring Ember.js~~ (moved to #823)
* [x] ~~Contributing to Ember.js~~ (I took a glance and it seems fine)

If you'd like to help out, just leave a comment with which section you'd like to take on.

### PR DESCRIPTION
**Pull request by acorncom**
Helps with #602

### CODE REVIEW

File: source/components/defining-a-component.md [hunk 1]
```diff
@@ -113,8 +110,8 @@ export default Ember.Route.extend({
 {{/each}}
 ```
 
-For brevity, `componentName` is hardcoded inside each post, but it can very
-well be a computed property that deduces the target component based on the data.
+In the example above `componentName` could be a simple string or a computed property
```
Comment (locks) [inline source/components/defining-a-component.md]: I'm not sure this bit is adding anything to the guide.

File: source/components/defining-a-component.md [hunk 2]
```diff
@@ -74,18 +74,15 @@ file at `app/components/blog-post.js`. If your component was called
 
 ### Dynamically rendering a component
 
-The `{{component}}` helper can be used to defer the selection of a component to
-run time. The `{{my-component}}` syntax would always render the same component,
-whereas using the `{{component}}` helper allows swapping the component rendered
-on the fly. This is useful in cases where, for example, you want to interact
-with different external libraries depending on the data. Using the `{{component}}`
-helper would allow you to keep those different logic well-separated.
+The `{{component}}` helper can be used to defer the selection of a component until later.
```
Comment (locks) [inline source/components/defining-a-component.md]: That's not as helpful, as "later" is very vague, while run time is a known thing.

File: source/components/defining-a-component.md [hunk 3]
```diff
@@ -74,18 +74,15 @@ file at `app/components/blog-post.js`. If your component was called
...
+The `{{my-component}}` syntax always renders the same component, while using the `{{component}}` helper allows choosing a component to render on the fly. This is useful in cases where
+you want to interact with different external libraries depending on the data. Using the `{{component}}` helper would allow you to keep different logic well-separated.
 
-The first parameter of the helper is the name of a component to render, as a string. So if you have `{{component 'blog-post'}}`, that is just the same as just `{{blog-post}}`.
+The first parameter of the helper is the name of a component to render, as a string. So  `{{component 'blog-post'}}`is just the same as using `{{blog-post}}`.
```
Comment (locks) [inline source/components/defining-a-component.md]: Missing space in post'}}`is.

Comment (acorncom) [inline source/components/defining-a-component.md]: @locks Missing or extra? If not extra, I think I'm missing something

File: source/components/defining-a-component.md (hunk 2 above)
Comment (acorncom) [inline source/components/defining-a-component.md]: Sounds good, I'll revert that. Was trying to make this easier to understand for the non-techie just picking up Ember

File: source/components/defining-a-component.md (hunk 1 above)
Comment (acorncom) [inline source/components/defining-a-component.md]: Yank the entire paragraph?

File: source/components/defining-a-component.md (hunk 3 above)
Comment (locks) [inline source/components/defining-a-component.md]: > So  `{{component 'blog-post'}}`is

to

> So `{{component 'blog-post'}}` is

File: source/components/defining-a-component.md (hunk 1 above)
Comment (locks) [inline source/components/defining-a-component.md]: Yeah, I think so. We can bring it back later if people stumble.

**Comment by acorncom** [comment]
@locks Revised

**Pull request merged**